### PR TITLE
samples: nrf9160: lwm2m_client: Set queue mode as default

### PR DESCRIPTION
**Pull request by juhaylinen**
- Remove overlay file for LwM2M queue mode and move the settings to prj.conf.
- Enable TAU pre-warning notifications.
- Update documentation.

### CODE REVIEW

**Review by jheiskan81 — APPROVED** [review]

File: samples/nrf9160/lwm2m_client/sample_description.rst [hunk 1]
```diff
@@ -460,14 +469,10 @@ After building and running the sample, you can locate your device in the server:
...
-==================
+Queue Mode
+==========
 
-To use the LwM2M Client with LwM2M Queue Mode support, build it with the ``-DOVERLAY_CONFIG=overlay-queue.conf`` option:
-
-.. code-block:: console
-
-   west build -b nrf9160dk_nrf9160_ns -- -DOVERLAY_CONFIG=overlay-queue.conf
+The sample enables LwM2M queue mode by default. In this mode, the device is not required to actively listen for incoming packets and the client can reduce power consumption by sleeping longer.
```
Comment (SeppoTakalo) [inline samples/nrf9160/lwm2m_client/sample_description.rst:475]:
```suggestion
The sample uses LwM2M queue mode by default. In this mode, the device is not required to actively listen for incoming packets and the client can reduce power consumption by sleeping longer.
```

A bit longer explanation on here should be given.
I think we should probably have a whole section of LTE network and Queue mode. But at least on this PR try to add for example following

> Queue mode is suitable for LTE connection where devices are not continuously reachable from Internet. Both, network and our modem try to release any resources as soon as possible.  In LTE the Radio Resource Control protocol (RRC) closes idle connections sometimes as soon as 10 seconds of idle time. This depends on the network operator.
> Another network dependent feature is the NAT timeout. Some networks drop unused UDP mappings after 30 seconds while RFC recommendation is 2 minutes. Therefore after a short sleeping period, our device is not addressable from Internet as the mapping does not exist anymore.
> Therefore our default configuration is that after contacting the LwM2M server, we only stay listening for 10 seconds after a last packet received. After that idle period, we try to enable eDRX or PSM power saving modes.
> Device wakes up from sleep, when it needs to send someting.

**Review by SeppoTakalo — CHANGES_REQUESTED** [review]
Left a review remark on documentation side.
Otherwise looks ok

File: samples/nrf9160/lwm2m_client/sample_description.rst (hunk 1 above)
Comment (juhaylinen) [reply samples/nrf9160/lwm2m_client/sample_description.rst]: Yes, I agree. I have now updated the documentation as suggested.

**Review by SeppoTakalo — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by umapraseeda — CHANGES_REQUESTED** [review]

**Comment by juhaylinen** [comment]
@umapraseeda  thanks for the review. I made the changes to documentation and added links for 3GPP TS 24.008

**Comment by umapraseeda** [comment]
Anything to add in the changelog? If not, the label changelog-entry-required can be removed.

**Review by umapraseeda — APPROVED** [review]

**Review by VeijoPesonen — APPROVED** [review]
Just minor things but otherwise looks good to me.

**Review by VeijoPesonen — APPROVED** [review]

**Review by SeppoTakalo — APPROVED** [review]

**Pull request merged**